### ISSUE TITLE
Improve boto_route53 execution module documentation

### ISSUE THREAD

**Issue by rallytime** [saltstack/salt#45143]
### Description of Issue/Question
The execution module documentation for `boto_route53.py` need some improvements. The states are documented well, but it's not clear how the `wait_for_sync` option can be used from the execution module.

This issue is to track documentation improvements proposed in #44949, which originated from issue #44947.

@tkwilliams I believe you volunteered to complete this effort. I would assign the issue to you, but you haven't accepted the `team-boto` invitation yet. Once that is done, I can assign the issue to you. Please let me know if you need a new invite. ;)

**Comment by tkwilliams** [comment]
@rallytime  Wow, I never realized I hadn't accepted the team-boto invitation.  Not sure where to look for it now - I don't see it in my email inbox.  Is it possible to re-send it, or alternatively is there somewhere inside github I can / should look?

**Comment by rallytime** [comment]
@tkwilliams I just cancelled your original invitation and sent a new one. I couldn't figure out how to "resend" it. Let me know if you still have trouble. You should be able to also see some kind of banner when you just go to the main saltstack repo page: https://github.com/saltstack/salt indicating you need to accept the invitation I think.

**Comment by tkwilliams** [comment]
Sweet!  Got the email, looks like I was able to join :)

**Comment by tkwilliams** [comment]
@rallytime When you added me back, did you flag me as a reviewer for team-boto related things?  I ask because I don't seem to have the "review" option on any of the tickets anymore (that I can find, anyway) :)

**Comment by rallytime** [comment]
@tkwilliams You should be getting notifications via the "team-boto" team through GitHub. Can you explain more to me about the "review" option for your tickets? You can't review your own PRs, but otherwise you should be able to review PRs like this one: https://github.com/saltstack/salt/pull/46290

**Comment by tkwilliams** [comment]
Sorry, I was being vague :)

Generally, when something "team-boto" shows up, I'll have a "Review this change" (or something close to it IIRC) option on the PR page...  That is no longer appearing on any of the tickets where it previously would have.  Now, that MAY be because at the moment all such tickets have already been reviewed / approved by either you or Ryan, but I seem to remember that even then I used to be able to add a review of my own.

E.g. on https://github.com/saltstack/salt/issues/45143 I added a plain comment a few minutes ago, because I don't see the option to review the change.
